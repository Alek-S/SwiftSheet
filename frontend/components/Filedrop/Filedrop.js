import React, { useState } from 'react';
import styled from 'styled-components';
import Dropzone from 'react-dropzone';
import PropTypes from 'prop-types';
import defaultStyle, { Card, ErrorDialog } from '../../defaultStyle';

/**
 * @function
 * Area for drag & drop file uploads of csv
 * @param {bool} - props.firstRowHeader - is first row a header
 * @returns {jsx} <Filedrop />
 */
const Filedrop = ({
	firstRowHeader,
	expireIn,
	setExpireIn,
	setHeader,
	password,
	setPassword,
	onDrop,
	wrongPassword,
}) => {
	const [dragging, setDragging] = useState(false);

	/**
	 * @function
	 * Set expireIn state, in Hrs
	 * @param {object} e - dom event
	 */
	const _handleChange = e => {
		e.preventDefault();
		setExpireIn(parseInt(e.target.value));
	};

	/**
	 * @function
	 * Toggle first row of CSV header boolean
	 * @param {object} e - dom event
	 */
	const _toggleHeader = e => {
		e.preventDefault();
		setHeader(!firstRowHeader);
	};

	return (
		<StyledSection>
			<StyledDropzone
				onDrop={onDrop}
				accept="text/csv"
				maxSize={5000000}
				minSize={1}
				multiple={false}
				dragging={dragging ? 'true' : undefined}
				onDragEnter={() => setDragging(true)}
				onDragLeave={() => setDragging(false)}
			>
				{dragging ? (
					<img
						src="./assets/images/upload-cloud-darkBlue.svg"
						alt="upload-cloud-logo"
					/>
				) : (
					<img
						src="./assets/images/upload-cloud-light.svg"
						alt="upload-cloud-logo"
					/>
				)}
				<StyledText dragging={dragging}>
					{dragging ? 'Drop' : 'Drag'} CSV file here
				</StyledText>
			</StyledDropzone>
			<Options role="form" aria-label="preferences">
				<section aria-label="set password">
					Password:
					<input
						value={password}
						className={wrongPassword ? wrongPassword.toString() : undefined}
						onChange={e => setPassword(e.target.value)}
						type="password"
						placeholder="enter password"
						autoComplete="new-password"
					/>
				</section>

				<section aria-label="set expiration">
					Expires In:
					<select value={expireIn} onChange={_handleChange}>
						<option value={1}>1 Hour</option>
						<option value={4}>4 Hours</option>
						<option value={8}>8 Hours</option>
						<option value={24}>1 Day</option>
						<option value={72}>3 Days</option>
						<option value={120}>5 Days</option>
					</select>
				</section>

				<section aria-label="toggle first row header">
					<HeaderToggle active={firstRowHeader}>
						<span>First Row Header:</span>
						<button onClick={_toggleHeader}>
							{firstRowHeader ? 'Yes' : 'No'}
						</button>
					</HeaderToggle>
				</section>
			</Options>

			<WrongPassword
				className={wrongPassword ? wrongPassword.toString() : undefined}
			>
				⚠️ Check Password Length
			</WrongPassword>
		</StyledSection>
	);
};

const StyledSection = styled.section`
	display: flex;
	flex-direction: column;
	align-items: center;
`;

const StyledDropzone = styled(Dropzone)`
	background-color: ${props =>
		props.dragging ? props.theme.color.blue : '#fff'};
	border: dashed 2px
		${props =>
			props.dragging ? props.theme.color.darkBlue : props.theme.color.border};
	border-radius: 5px;
	height: 250px;
	width: 70%;
	margin: auto;
	margin-top: 3rem;

	img {
		display: block;
		height: 200px;
		width: 80%;
		margin: auto;
	}
`;

const StyledText = styled(defaultStyle)`
	color: ${props =>
		props.dragging ? props.theme.color.darkBlue : props.theme.color.lightText};
	font-size: 1.2rem;
	text-align: center;
	background-color: transparent;
`;

const Options = styled(Card)`
	padding-top: 1.75rem;
	padding-bottom: 1.75rem;

	display: flex;
	flex-direction: row;
	flex-wrap: wrap;
	justify-content: space-evenly;
	align-items: center;
	width: 70%;
	margin: auto;
	text-align: center;
	margin-top: 3.5rem;

	input {
		margin: 0 2rem 0 0.25rem;
		padding-left: 1rem;
		width: 140px;
		font-size: 0.9rem;
		font-weight: 400;
		height: 25px;

		::placeholder {
			color: ${props => props.theme.color.backgroundDarkest};
		}
	}

	select {
		margin-left: 0.25rem;
	}
`;

const HeaderToggle = styled.label`
	margin-left: 2rem;

	button {
		color: white;
		cursor: pointer;
		outline: none;
		background: ${props =>
			props.active
				? props.theme.gradient.greenBlue
				: props.theme.color.backgroundDarkest};
		box-shadow: ${props => (props.active ? props.theme.boxShadow : '')};
		border: none;
		font-family: ${props => props.theme.font.main};
		font-size: 1rem;
		border-radius: 4px;
		margin-left: 1rem;
		padding-top: 5px;
		padding-bottom: 5px;
		width: 70px;
	}
`;

Filedrop.propTypes = {
	firstRowHeader: PropTypes.bool,
	expireIn: PropTypes.number,
	setExpireIn: PropTypes.func,
	setHeader: PropTypes.func,
	password: PropTypes.string,
	setPassword: PropTypes.func,
	onDrop: PropTypes.func,
};

const WrongPassword = styled(ErrorDialog)`
	width: 200px;
	margin-top: 2rem;
`;

export default Filedrop;
