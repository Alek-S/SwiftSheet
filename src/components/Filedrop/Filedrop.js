import React, { Component } from 'react';
import styled from 'styled-components';
import Dropzone from 'react-dropzone';
import Papa from 'papaparse';
import defaultStyle from '../../defaultStyle';

/** @class
 * @name Filedrop
 * Area for drag & drop file uploads of csv
 *
 * @returns {JSX}
 */
export default class Filedrop extends Component {
	constructor(props) {
		super(props);
		this.state = {
			file: null,
			rejectedFile: [],
			dragging: false,
		};
	}

	onDrop = (file, rejectedFile) => {
		this.setState({ file, rejectedFile });
		Papa.parse(file[0], {
			header: false,
			download: true,
			skipEmptyLines: false,
			complete: this.handleJSONData,
		});
	};

	handleJSONData = result => {
		const data = result.data;
		console.log(data);
	};

	onDragEnter = () => this.setState({ dragging: true });

	onDragLeave = () => this.setState({ dragging: false });

	render() {
		const { dragging } = this.state;

		return (
			<section style={{ height: 'calc(100vh - 100px)' }}>
				<StyledDropzone
					onDrop={this.onDrop}
					accept="text/csv"
					maxSize={5000000}
					minSize={1}
					multiple={false}
					dragging={dragging}
					onDragEnter={this.onDragEnter}
					onDragLeave={this.onDragLeave}
				>
					<StyledText>{dragging ? 'Drop' : 'Drag'} CSV file here</StyledText>
				</StyledDropzone>
			</section>
		);
	}
}

const StyledDropzone = styled(Dropzone)`
	background-color: ${props =>
		props.dragging ? props.theme.color.blue : '#fff'};
	border: dashed 2px
		${props =>
			props.dragging ? props.theme.color.darkBlue : props.theme.color.border};
	border-radius: 5px;
	height: 250px;
	width: 60%;
	margin: auto;
	margin-top: 4rem;
`;

const StyledText = styled(defaultStyle)`
	font-size: 1.2rem;
	text-align: center;
	background-color: transparent;
	padding-top: 7rem;
`;
