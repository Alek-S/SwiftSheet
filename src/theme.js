/**
 * default themes for use with styled-components
 * passed in with theme provider
 */
const theme = {
	boxShadow: `0px 5px 40px -5px rgba(0,0,0,0.3)`,
	boxShadowDark: `0px 5px 40px -2px rgba(0,0,0,0.5)`,
	color: {
		background: '#fafafa',
		backgroundDark: '#EAEAEA',
		backgroundDarkest: '#BCBCBC',
		border: '#d9d9d9',
		text: '#404040',
		lightText: '#848484',
		red: '#ED5050',
		lightRed: '#FF8D8D',
		green: '#D8E13F',
		blue: '#46C8F5',
		lightBlue: '#97E4FF',
		darkBlue: '#2E8AA8',
	},
	gradient: {
		light: `linear-gradient(to bottom, #ffffff 0%, #f6f6f6 47%, #ededed 100%)`,
		red: 'linear-gradient(to bottom, #f07878 0%, #ED5050 100%)',
		lightRed: 'linear-gradient(to bottom, #f39191 0%, #ef6262 100%)',
		blue: 'linear-gradient(to bottom, #D8E13F 0%, #BFC644 100%)',
		greenBlue: ' linear-gradient(160deg, #57DACB 0%, #55A2ED 100%)',
	},
	font: {
		header: `'Poiret One', cursive`,
		main: `'Quicksand', sans-serif`,
	},
};

export default theme;
