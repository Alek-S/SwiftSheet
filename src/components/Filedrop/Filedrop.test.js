import React from 'react';
import Filedrop from './Filedrop';
import theme from '../../theme';


describe('Filedrop', () => {
	it('matches the snapshot', () => {
		const tree = shallowWithTheme(<Filedrop />, theme);
		expect(toJson(tree)).toMatchSnapshot();
	});
});