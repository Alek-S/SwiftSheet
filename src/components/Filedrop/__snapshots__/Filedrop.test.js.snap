// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filedrop matches the snapshot 1`] = `
<section
  style={
    Object {
      "height": "calc(100vh - 100px)",
    }
  }
>
  <Filedrop__StyledDropzone
    accept="text/csv"
    dragging={false}
    maxSize={5000000}
    minSize={1}
    multiple={false}
    onDragEnter={[Function]}
    onDragLeave={[Function]}
    onDrop={[Function]}
  >
    <Filedrop__StyledText>
      Drag
       CSV file here
    </Filedrop__StyledText>
  </Filedrop__StyledDropzone>
</section>
`;
